ee/query-service/app/server.go

```diff
@@ -266,15 +269,82 @@ func (lrw *loggingResponseWriter) Flush() {
 	lrw.ResponseWriter.(http.Flusher).Flush()
 }
 
+func extractDashboardMetaData(path string, r *http.Request) (map[string]interface{}, bool) {
+	pathToExtractBodyFrom := "/api/v2/metrics/query_range"
+	var requestBody map[string]interface{}
+	data := map[string]interface{}{}
+
+	if path == pathToExtractBodyFrom && (r.Method == "POST") {
+		bodyBytes, _ := ioutil.ReadAll(r.Body)
+		r.Body.Close() //  must close
+		r.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+
+		json.Unmarshal(bodyBytes, &requestBody)
+
+	} else {
+		return nil, false
+	}
+
+	compositeMetricQuery, compositeMetricQueryExists := requestBody["compositeMetricQuery"]
+	compositeMetricQueryMap := compositeMetricQuery.(map[string]interface{})
```

Comment:
It's not failing today because the frontend is sending dummy `compositeMetricQuery` data for promql/ClickHouse queries, but otherwise, I believe this would fail with a nil pointer when `compositeMetricQueryExists` is false. Please make sure not to panic anywhere in the telemetry collection process. Please move this inside below if block.

Reply:
right, let me handle that in another PR

Reply:
okay